File: even_sum/even_sum.py
def accumSum(l, n):
    if(len(l) == 0):
        return n
    else:
        x, *xs = l
        if(x % 2 == 0):
            return accumSum(xs, x + n)
        else:
            return accumSum(xs, n)

def evenSum(l):
    return accumSum(l, 0)

print(evenSum([1, 2, 3, 4, 5]))

File: code/even_sum/even_sum.hs
foldl' :: (a -> b -> a) -> a -> [b] -> a
foldl' f a []     = a
foldl' f a (x:xs) = seq a' $ foldl' f a' xs
    where a' = f a x

evenSum l = mysum $ filter even l
       where mysum []     = 0
             mysum (x:xs) = x + mysum xs
